llarp/router/router.cpp

```diff
@@ -1004,22 +998,22 @@ namespace llarp
       return false;
     });
 
-    // find all deregistered relays
-    std::unordered_set<PubKey> close_peers;
-
-    for_each_connection([this, &has_whitelist, &close_peers](link::Connection& conn) {
-      if (follow_whitelist && has_whitelist)
-        return;
+    if (not is_snode or not has_whitelist)
+    {
+      // find all deregistered relays
+      std::unordered_set<PubKey> close_peers;
 
-      const auto pk = conn.remote_rc.pubkey;
+      for_each_connection([this, &close_peers](link::Connection& conn) {
+        const auto& pk = conn.remote_rc.pubkey;
 
-      if (conn.remote_is_relay and not _rc_lookup_handler.is_session_allowed(pk))
-        close_peers.emplace(pk);
-    });
+        if (conn.remote_is_relay and not _rc_lookup_handler.is_session_allowed(pk))
+          close_peers.insert(pk);
```

Comment:
insert instead of emplace because it's a const reference to an already a constructed object, right?

Would using emplace copy construct a new object? Or would emplace just copy-elision it into the unordered set?